src/gleam/result.gleam

```diff
@@ -166,6 +176,12 @@ pub fn then(
   }
 }
 
+/// DEPRECATED: Use `result.try` instead.
+///
+pub fn then(result, fun) {
```

Comment:
Could you add the type annotations too please?

Reply:
This was intentional so that it infers from the function it proxies. I will readd the types NP.